science/platform-implementations/scd40.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <libhal-util/i2c.hpp>
+#include <libhal-util/serial.hpp>
+#include <libhal-util/steady_clock.hpp>
+#include <libhal/units.hpp>
+
+
+//TODO: maybe later add CRC and config for altitude, ambient pressure, temperature offset (or look at setting persistent values)
+
+namespace sjsu::science {
+class scd40
+{
+private:
+
+    scd40(hal::i2c& p_i2c, hal::steady_clock& p_clock);
+    hal::i2c& m_i2c;
+    hal::steady_clock& m_clock;
+    hal::byte generate_crc(std::array<hal::byte, 2> data);
+    bool validate_crc(std::array<hal::byte, 3> data);
+
+    
+    //TODO: revise output type
+    enum addresses {// deal with hal::byte later 
```

Comment:
```suggestion
    enum class addresses { // deal with hal::byte later 
```

Great work using an enum for all of the addresses. Makes it easier to understand whats what. But its favorable to use an `enum class` as they get promoted to a type. You can `static_cast<>` them back to whatever integral repersation you need provided that representation does not truncate the enum value. You can also use `hal::value()` in `<libhal-util/enum.hpp>`

Reply:
I talked to Corey and Gene to understand what you were saying. While on the topic they recommended using constants (likely in a separate file) for this use case. What are your thoughts on switching the values in the enum to constants?

Reply:
static constants are great! Sometimes the constructs of an enum work fine too just depends on what you are doing.